lib/JSONSchemaGenerator.js

```diff
@@ -245,47 +269,86 @@ module.exports = class JSONSchemaGenerator {
       name: collectionName,
       defaultState: collectionDefaultState,
     } = collectionDefinition
-    this._pathFieldsRawSchema = pathFieldsRawSchema
-    this._collectionName = collectionName
-    this._collectionDefinition = collectionDefinition
-    this._collectionBasePath = formatEndpointTag(endpointBasePath)
 
     const idType = getIdType(collectionDefinition)
-    this._idType = idType
-    this._requiredFields = getRequiredFields(collectionDefinition)
+    const {
+      collectionFields: validationProperties,
+      collectionRawObject: patternProperties,
+      required,
+      sort,
+      changeStateProperties,
+      collectionFieldsArrayOperations,
+      unsetPatternProperties,
+      unsetProperties,
+      dateProperties,
+    } = collectionFieldsProperties(collectionDefinition, true)
 
-    this._serializationProperties = collectionFieldsProperties(structuredClone(collectionDefinition), false)
-    this._propertiesGetListValidation = propertiesGetListValidation(
-      structuredClone(collectionDefinition),
+    const {
+      collectionFields: serializationProperties,
+    } = collectionFieldsProperties(collectionDefinition, false)
```

Comment:
Since the `collectionFieldsProperties` is now called only here, we might want to consider, in the future, to remove the `validation` flag passed as parameter to call this method only once (and maybe renaming `collectionFields` and `collectionRawObject`, for further clarity).

Reply:
We could, but we have to consider that in serialization we have different type conversion (for `Date` fields). Should I try to do it now?

Reply:
(sorry for the very late answer)

No, no need to rush. It's fine, especially if we consider to use serialization for further uses. Just let's keep this in mind in future developments.

Reply:
Maybe with validation modification I'm developing in another branch we will have a unique behavior for validation and serialization. In this case we could also add this improvement